src/compose/service.ts

```diff
@@ -56,6 +54,13 @@ export class Service {
 		'groupAdd',
 		'securityOpt',
 	];
+	public static orderedConfigArrayFields: ServiceConfigArrayField[] = [
+		'dns',
+		'dnsSearch',
```

Comment:
What happens if `devices` contains an overlap in container paths? For instance,
```
devices:
  - "/dev/device-1:/dev/device-1"
  - "/dev/device-2:/dev/device-1"
```

Does it result in an error or does the order matter?

Reply:
We only need to worry about order when the order has some meaning, and the docker daemon won't change it when reporting the current state. So the above shouldn't be an issue.

Reply:
My question was basically if order matters for `devices:`. And a quick test shows it does.
```
romans-mbp:resin-ui roman$ docker run --device=/dev/zero:/dev/test1 --device=/dev/urandom:/dev/test1  -it --rm busybox
/ # cat /dev/test1
^C
```

Here, `cat` does not produce any output, so it means `/dev/zero` got mapped to `/dev/test1` in the container. So order of `--device` matters, and container has to be recreated if elements there are swapped...

So, should we add `devices` to this array too?

Reply:
Ah sorry I misunderstood your original point - this is actually a different instance of the order being important, as something is being overwritten as it were.  This actually bears more thinking about, as I believe the volumes could also suffer from this problem, if you mount something on the same mountpoint. This is even more confusing as the volumes returned from docker can definitely be in a different order to being referenced. 

Given that exists as a problem now, I'd rather think about this as a whole with all of the possible ways that this could fail, and merge this now to at least support `dns` and `dnsSearch` properly, as this will not have the same problem.